autorest/adal/token.go

```diff
@@ -1317,12 +1317,28 @@ func NewMultiTenantServicePrincipalTokenFromCertificate(multiTenantCfg MultiTena
 
 // MSIAvailable returns true if the MSI endpoint is available for authentication.
 func MSIAvailable(ctx context.Context, s Sender) bool {
+	_, _, err := getMSIAvailable(ctx, s)
```

Comment:
With the caveat that I'm not familiar with the usage patterns around this function and may be missing some context, I would have this function simply return `true` when MSI_ENDPOINT is set, in order to:
1. trust the platform to configure the environment
2. surface App Service problems as authentication errors rather than swallowing them here
3. probe only the IMDS endpoint, if we probe at all, because it's the only one with no availability indicator

Reply:
Sounds good to me! I've just updated the PR now 👍🏼